File: cropme/src/main/java/com/takusemba/cropme/MoveAnimator.java
package com.takusemba.cropme;

import android.support.animation.SpringForce;

/**
 * interface to move Image.
 **/
interface MoveAnimator {

    /**
     * stiffness when flinging or bouncing
     **/
    float STIFFNESS = SpringForce.STIFFNESS_VERY_LOW;

    /**
     * dumping ratio when flinging or bouncing
     **/
    float DAMPING_RATIO = SpringForce.DAMPING_RATIO_NO_BOUNCY;

    /**
     * friction when flinging
     **/
    float FRICTION = 3f;

    /**
     * move image
     *
     * @param delta distance of how much image moves
     **/
    void move(float delta);

    /**
     * bounce image when image is off of {@link CropOverlay#getFrame()}
     *
     * @param velocity velocity when starting to move
     **/
    void reMoveIfNeeded(float velocity);

    /**
     * fling image
     *
     * @param velocity velocity when starting to fling
     **/
    void fling(float velocity);

    /**
     * true if image is flinging, false otherwise
     **/
    boolean isNotFlinging();

}

File: cropme/src/main/java/com/takusemba/cropme/internal/MoveAnimator.kt
package com.takusemba.cropme.internal

import androidx.dynamicanimation.animation.SpringForce

/**
 * Interface to move Image.
 */
internal interface MoveAnimator {

  /**
   * Move image
   *
   * @param delta distance of how much image moves
   */
  fun move(delta: Float)

  /**
   * adjust image when image is off the frame
   */
  fun adjust()

  /**
   * fling image
   *
   * @param velocity velocity when starting to fling
   */
  fun fling(velocity: Float)

  companion object {

    /**
     * stiffness when flinging or bouncing
     */
    const val STIFFNESS = SpringForce.STIFFNESS_VERY_LOW

    /**
     * dumping ratio when flinging or bouncing
     */
    const val DAMPING_RATIO = SpringForce.DAMPING_RATIO_NO_BOUNCY

    /**
     * friction when flinging
     */
    const val FRICTION = 3f
  }
}
